jwcrypto/jwe.py

```diff
@@ -289,6 +290,56 @@ def unwrap(self, key, keylen, ek, headers):
         return cek
 
 
+class _Pbes2HsAesKw(_RawKeyMgmt):
+
+    def __init__(self, hashsize, keysize):
+        self.backend = default_backend()
+        self.hashsize = hashsize
+        self.keysize = keysize // 8
+
+    def _get_key(self, alg, key, p2s, p2c):
```

Comment:
I would have to add a non-standard Key type to JWK to do that, because the JWE interface requires that key is a JWK object.
Alternatively I'd need to allow to pass in a string just for the PBES2 algorithm and forgo initial validation of the input ...

anyway see PR #40